internal/api/parse.go

```diff
@@ -76,15 +76,16 @@ func (p Parser) parseNestedConnect(parentBinding db.Binding, data map[string]int
 	if len(data) != 1 {
 		panic("Too many keys in a unique query")
 	}
-	m, err := p.tx.GetModelById(parentBinding.Dual().ModelId())
+	m, err := p.tx.GetModelByID(parentBinding.Dual().ModelID())
 	if err != nil {
 		return
 	}
 	// this should be a separate method
 	var uq UniqueQuery
 	for k, v := range data {
 		var val interface{}
-		val, err = m.AttributeByName(k).ParseFromJson(v)
+		d := m.AttributeByName(k).Datatype
+		val, err = db.CallFunc(d.Validator, d.StorageFormat, v)
```

Comment:
Two thoughts here.

First, I think this is exposing some implementation details of FromJson (that it uses potentially custom code) that ideally the Parser wouldn't have to know about (it's got nothing to do with datatype parsing -- it's about api operation parsing).

The solution to this one would be encapsulating the call inside of Datatype (ie have a method Datatype.FromJson(v) that internally uses db.CallFunc. 

Secondly, CallFunc doesn't really seem to me like it belongs in the db package. But I get that it's tricky: It uses concepts that are specific to itself (the Code model and Runtimes) that depend on db, and the db package depends on it in the form of using Datatypes.

The way to cut that knot is to separate out the core idea of a runtime and executing some code in a given runtime from the db-dependent Code object.

So for instance, we might have a module that encapsulates the idea of runtimes and invoking functions that exposes a service object (like how db.DB is a service object), "executor" that exposes a method "execute" that takes the relevant components of a Code without taking the actual db.Record itself..

Reply:
Got it. Will submit for review soon!